src/Compilers/Core/Portable/StrongName/StrongNameFileSystem.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class StrongNameFileSystem
+    {
+        internal virtual byte[] ReadAllBytes(string fullPath)
+        {
+            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
+            return File.ReadAllBytes(fullPath);
+        }
+
+        /// <summary>
+        /// Resolves assembly strong name key file path.
+        /// </summary>
+        /// <returns>Normalized key file path or null if not found.</returns>
+        internal string ResolveStrongNameKeyFile(string path, ImmutableArray<string> keyFileSearchPaths)
+        {
+            // Dev11: key path is simply appended to the search paths, even if it starts with the current (parent) directory ("." or "..").
+            // This is different from PathUtilities.ResolveRelativePath.
+
+            if (PathUtilities.IsAbsolute(path))
+            {
+                if (FileExists(path))
+                {
+                    return FileUtilities.TryNormalizeAbsolutePath(path);
+                }
+
+                return path;
+            }
+
+            foreach (var searchPath in keyFileSearchPaths)
+            {
+                string combinedPath = PathUtilities.CombineAbsoluteAndRelativePaths(searchPath, path);
+
+                Debug.Assert(combinedPath == null || PathUtilities.IsAbsolute(combinedPath));
+
+                if (FileExists(combinedPath))
+                {
+                    return FileUtilities.TryNormalizeAbsolutePath(combinedPath);
+                }
+            }
+
+            return null;
+        }
+
+        internal virtual bool FileExists(string fullPath)
+        {
+            Debug.Assert(fullPath == null || PathUtilities.IsAbsolute(fullPath));
+            return File.Exists(fullPath);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+
+            return true;
+        }
```

Comment:
If we're implementing `Equals`, we'll need to implement `GetHashCode` as well. I'd rather we implement neither though.

Reply:
`return 0` it is.